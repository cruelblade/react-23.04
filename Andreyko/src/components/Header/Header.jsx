import React, { Component } from 'react';
import PropTypes from "prop-types";
import { Link } from 'react-router-dom';

import './style.css'

export default class Header extends Component {
  render() {
    const { chatId } = this.props;
    return (
      <div className="header-container">
        <h1 className="chat-heading">{`Чат ${chatId}`}</h1>
        <Link to="/profile/">
          <h2>Мой профиль</h2>
        </Link>
      </div>
    )
  }
}

Header.propTypes = {
  chatId: PropTypes.number,
};

Header.defaultProps = {
  chatId: 1,
};