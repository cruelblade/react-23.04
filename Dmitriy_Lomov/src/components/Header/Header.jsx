import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';

import cx from 'classnames';
import { AppBar, Toolbar, Typography, withStyles } from '@material-ui/core';

const muiStyles = theme => ({
  toolbar: {
    paddingRight: 24,
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: 240,
    width: `calc(100% - 240px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  menuButtonHidden: {
    display: 'none',
  },
  title: {
    flexGrow: 1,
  },
});

class Header extends PureComponent {
  render() {
    const { profile, classes } = this.props;

    return (
      <AppBar position="absolute" className={cx(classes.appBar, classes.appBarShift)}>
        <Toolbar className={classes.toolbar}>
          <Typography
            component="h1"
            variant="h6"
            color="inherit"
            noWrap
            className={classes.title}
          >
            Эмулятор чата
          </Typography>
          <Typography
            component="p"
            variant="body2"
            color="textSecondary"
            noWrap
            className={classes.title}
          >
            Текущий профиль: {profile.name} {profile.lastName}
          </Typography>
        </Toolbar>
      </AppBar>
    );
  }
}

const mapStateToProps = store => ({
  profile: store.profile,
});

export default compose(connect(mapStateToProps), withStyles(muiStyles))(Header);
