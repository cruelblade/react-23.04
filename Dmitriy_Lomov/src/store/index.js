import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';
import notification from './middleware/notification';

const logger = createLogger({ collapsed: true });

const store = createStore(rootReducer, applyMiddleware(thunk, logger, notification));

export default store;
