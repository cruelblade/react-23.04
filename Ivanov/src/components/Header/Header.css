.root {
    display: flex;
    z-index: 2;
}

.title {
    flex-grow: 1;
}