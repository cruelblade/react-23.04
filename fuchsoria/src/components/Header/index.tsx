import React from 'react';
import { Menu } from 'antd';
import { useHistory } from 'react-router-dom';
import styles from './styles.module.scss';

type TypeSwitchPage = { key: string; keyPath?: Array<string>; item?: unknown; domEvent?: Event };

export default function Header() {
  const history = useHistory();
  const defaultSelected = history?.location?.pathname.split('/')[1];
  const switchPage = ({ key }: TypeSwitchPage) => {
    history.push(`/${key}`);
  };

  return (
    <div className={styles.headerContainer}>
      <h2 className={styles.headerLogo}>Messenger</h2>
      <Menu theme="dark" mode="horizontal" defaultSelectedKeys={[defaultSelected || 'home']} onClick={switchPage}>
        <Menu.Item key="home">Home</Menu.Item>
        <Menu.Item key="chats">Chats</Menu.Item>
        <Menu.Item key="profile">Profile</Menu.Item>
      </Menu>
    </div>
  );
}
