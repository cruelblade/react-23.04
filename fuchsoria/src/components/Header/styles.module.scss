.header {

  &__container {
    display: flex;
    justify-content: space-between;
  }
  
  &__logo {
    height: 64px;
    padding: 10px 20px;
    color: #fff;
    font-size: 36px;
    line-height: 1;
    cursor: pointer;
    transition: opacity 0.3s;

    &:hover {
      opacity: 0.7;
    }
  }
}
