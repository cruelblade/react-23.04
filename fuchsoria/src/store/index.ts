import { fetchMiddleware } from './middlewares/fetchMiddleware';
import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import { composeWithDevTools } from 'redux-devtools-extension';
import storage from 'redux-persist/lib/storage';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { apiMiddleware } from 'redux-api-middleware';
import chatsReducer from './reducers/chatsReducer';
import profileReducer from './reducers/profileReducer';
import chatlistReducer from './reducers/chatlistReducer';
import { botAnswer } from './middlewares/botMiddleware';
import { autoRedirect } from './middlewares/chatMiddleware';
import { createBrowserHistory, History } from 'history';

const history = createBrowserHistory();

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const reducer = (history: History) =>
  combineReducers({
    chats: chatsReducer,
    profile: profileReducer,
    chatlist: chatlistReducer,
    router: connectRouter(history),
  });

const persistConfig = {
  key: 'messengerweb',
  whitelist: ['chats', 'profile'],
  storage,
};

export function initStore(preloadedState = undefined) {
  const persistedReducer = persistReducer(persistConfig, reducer(history));

  const middlewares = [routerMiddleware(history), apiMiddleware, fetchMiddleware, thunk, botAnswer, autoRedirect];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(persistedReducer, preloadedState, composedEnhancers);
  const persistor = persistStore(store);

  return { store, persistor, history };
}
