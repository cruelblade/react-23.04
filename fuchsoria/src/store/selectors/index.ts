import { IChats, IProfileState } from './../../interfaces';

export const selectChats = (state: { chats: IChats }) => {
  const { chats } = state;

  return chats;
};

export const selectProfile = (state: { profile: IProfileState }) => {
  const { profile } = state;

  return profile;
};
