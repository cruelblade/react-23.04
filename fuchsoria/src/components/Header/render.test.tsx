import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initStore } from '../../store';
import Header from '../Header';

const { store } = initStore();

describe('<Header />', () => {
  it('Component should render', () => {
    const { container } = render(
      <Provider store={store}>
        <Header />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
