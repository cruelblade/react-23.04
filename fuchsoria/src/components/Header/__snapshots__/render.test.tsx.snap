// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Component should render 1`] = `
<div>
  <div
    class="headerContainer"
  >
    <h2
      class="headerLogo"
    >
      Messenger
    </h2>
    <ul
      class="ant-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
      role="menu"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
        role="menuitem"
      >
        Home
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        Chats
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        Profile
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </div>
</div>
`;
